Makefile

```diff
@@ -0,0 +1,72 @@
+# opentrons modules makefile
+# https://github.com/Opentrons/opentrons-modules
+
+SHELL := /bin/bash
+
+LIBRARIES_DIR := libraries
+MODULES_DIR := modules
+INO_DIR := $(HOME)/arduino_ide
+
+ARDUINO_VERSION ?= 1.8.5
+CI_SYSTEM ?= linux
+
+ifeq ($(CI_SYSTEM), osx)
+	ARDUINO := $(INO_DIR)/Arduino.app/Contents/MacOS/Arduino
+	ARDUINO_ARCHIVE := arduino-$(ARDUINO_VERSION)-macosx.zip
+	UNPACK_CMD := unzip $(ARDUINO_ARCHIVE) -d $(HOME)/arduino_ide/
+	ARDUINO_SKETCHBOOK_LOC := $(HOME)/Documents/Arduino
+	ARDUINO15_LOC := $(HOME)/Library/Arduino15
+else
+	ARDUINO := $(INO_DIR)/arduino
+	ARDUINO_ARCHIVE := arduino-$(ARDUINO_VERSION)-linux64.tar.xz
+	UNPACK_CMD := tar xf $(ARDUINO_ARCHIVE) -C $(HOME)/arduino_ide/ --strip-components=1
+	ARDUINO_SKETCHBOOK_LOC := $(HOME)/Arduino
+	ARDUINO15_LOC := $(HOME)/.arduino15
+endif
+
+BUILD_PATH := $(HOME)/.arduino_builds
+DOWNLOAD_LINK := https://downloads.arduino.cc/$(ARDUINO_ARCHIVE)
+
+ifeq ($(wildcard $(ARDUINO)), )
+	NO_ARDUINO := true
+endif
+
+# setup Arduino
+.PHONY: setup-ino
+setup-ino:
```

Comment:
Nitpick, but GNU (and also to an extend, our) convention would probably be to call this task `install`.

Something that we've (recently) found to be helpful is to also have an `uninstall` task that undoes whatever `install` does. In this case, it would wipe out `$(HOME)/arduino_ide` I think

Reply:
@sfoster1 thoughts on `install`? I think if we're going to use this to set up arduino on our computers and build firmware without using the IDE then the `install` convention doesn't seem too bad..

Reply:
Per https://www.gnu.org/prep/standards/html_node/Standard-Targets.html , the `install` target is for installing _the program the makefile is meant to produce_. GNU doesn't have an equivalent for our usage of install, which is "install the dependencies required to build the target" - usually GNU autotools projects would discover missing dependencies in the `configure` step, complain about it, and trust the user to fix it.

Consider that muscle memory autotools sequence: `./configure && make && [sudo] make install`. When we use `install` to install dependencies, we're using it in a totally different way.

/rant

Reply:
@sfoster1 @sanni-t our current `install` usage is definitely a convention I've either introduced or pushed strongly. My bad for somehow entirely skipping over the "install" section of the standard targets doc every time I've read it and going straight to what my brain wants `make install` to be based on my experience with `npm install`.

I'm fine calling what we currently call `install` and `uninstall` something else. `setup` and `teardown` come to mind?

Reply:
`setup` and `teardown` sounds great!

Reply:
Cool